### PR TITLE
[Mediaite] Broken downloading fix

### PR DESCRIPTION
**Pull request by Grabien**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

This is a fix for downloading content from the Mediaite website. This site has changed the format of the JWPlayer code. The ID should now be extracted from JWPlyer links like this:

`script src="https://cdn.jwplayer.com/players/dtUIL6fD-qWJ2WQa4.js`

or

`data-video-adfree="https://cdn.jwplayer.com/players/nwpt1elX-tv6LAqIf.html`

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details><details><summary>Copilot Summary</summary>  

### Summary
🆕🛠️📺

Add a new test case and update the regex for the `Mediaite` extractor in `mediaite.py`. This improves the extractor's ability to handle different URL formats.

> _`Mediaite` tests_
> _New URL format added_
> _Winter of bugs ends_

### Walkthrough
* Add and modify regex patterns to extract video id from different URL formats ([link](https://github.com/yt-dlp/yt-dlp/pull/7923/files?diff=unified&w=0#diff-f28b59d426f8c68b7fd5fee78042c79098ed1af377f074c3df2bb07dd596a052L84-R101),                             F0L

</details>

### CODE REVIEW

File: yt_dlp/extractor/mediaite.py [hunk 1]
```diff
@@ -81,10 +81,23 @@ class MediaiteIE(InfoExtractor):
...
+            'upload_date': '20230802',
+        },
+        'params': {'skip_download': True}
     }]
 
     def _real_extract(self, url):
         webpage = self._download_webpage(url, None)
-        id = self._search_regex(r'data-video-id\s?=\s?\"([^\"]+)\"', webpage, 'id')
+        id = self._search_regex(r'"https://cdn\.jwplayer\.com/players/(.*)\.', webpage, 'id').split('-')[0]
         data_json = self._download_json(f'https://cdn.jwplayer.com/v2/media/{id}', id)
```
Comment (bashonly) [inline yt_dlp/extractor/mediaite.py:102]: Avoid `.*` in regex when possible
and while we are fixing the extractor, let's make it so it's not using a builtin as a var name
```suggestion
        video_id = self._search_regex(
            r'<script[^>]+src="https://cdn\.jwplayer\.com/players/(\w+)', webpage, 'id')
        data_json = self._download_json(f'https://cdn.jwplayer.com/v2/media/{video_id}', video_id)
```

**Review by bashonly — CHANGES_REQUESTED** [review]

**Review by bashonly — APPROVED** [review]

File: yt_dlp/extractor/mediaite.py [hunk 2]
```diff
@@ -81,10 +81,23 @@ class MediaiteIE(InfoExtractor):
             'upload_date': '20210930',
         },
         'params': {'skip_download': True}
+    }, {
+        'url': 'https://www.mediaite.com/politics/i-cant-read-it-fast-enough-while-defending-trump-larry-kudlow-overwhelmed-by-volume-of-ex-presidents-legal-troubles/',
+        'info_dict': {
+            'id': 'E6EhDX5z',
+            'ext': 'mp4',
+            'title': "Fox Business Network - 4:00 PM - 5:00 PM - 1:39:42 pm - 1:42:20 pm",
```
Comment (pukkandan) [inline yt_dlp/extractor/mediaite.py:89]:
```suggestion
            'title': 'Fox Business Network - 4:00 PM - 5:00 PM - 1:39:42 pm - 1:42:20 pm',
```

**Review by pukkandan — APPROVED** [review]

**Comment by pukkandan** [comment]
Consider falling back to the old old regex just in case. Or, make sure to double check all old tests

**Comment by Grabien** [comment]
I added a fallback regex with the old pattern. But the plugin passes all tests only with the new pattern, and I can't find the old pattern on any page.

File: yt_dlp/extractor/mediaite.py [hunk 3]
```diff
@@ -81,10 +81,26 @@ class MediaiteIE(InfoExtractor):
...
     }]
 
     def _real_extract(self, url):
         webpage = self._download_webpage(url, None)
-        id = self._search_regex(r'data-video-id\s?=\s?\"([^\"]+)\"', webpage, 'id')
+        id = self._search_regex(
+            [r'"https://cdn\.jwplayer\.com/players/(.*)-',
+             r'data-video-id\s?=\s?\"([^\"]+)\"'],
+            webpage, 'id')
         data_json = self._download_json(f'https://cdn.jwplayer.com/v2/media/{id}', id)
```
Comment (bashonly) [inline yt_dlp/extractor/mediaite.py:105]:
```suggestion
        video_id = self._search_regex(
            [r'"https://cdn\.jwplayer\.com/players/(\w+)', r'data-video-id\s*=\s*\"([^\"]+)\"'], webpage, 'id')
        data_json = self._download_json(f'https://cdn.jwplayer.com/v2/media/{video_id}', video_id)
```

**Pull request merged**